tests/lib/rules/no-unused-vars.js

```diff
@@ -995,6 +995,27 @@ ruleTester.run("no-unused-vars", rule, {
                 definedError("c")
             ]
         },
+        {
+            code: `let x = 0;
+            x++; x = 0;`,
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [{ ...assignedError("x"), line: 2, column: 18 }]
+        },
+
+        // https://github.com/eslint/eslint/issues/14325
+        {
+            code: `let x = 0;
+            x++, x = 0;`,
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [{ ...assignedError("x"), line: 2, column: 18 }]
+        },
+        {
+            code: `let x = 0;
+            x++, x = 0;
+            x=3;`,
+            parserOptions: { ecmaVersion: 2015 },
+            errors: [{ ...assignedError("x"), line: 3, column: 13 }]
+        },
```

Comment:
Can you also add these `invalid` tests: 

`let x = 0; x++, 0;`
`let x = 0; 0, x++;`
`let x = 0; 0, (1, x++);`
`let x = 0; foo = (x++, 0);`
`let x = 0; foo = ((0, x++), 0);`

Reply:
Done 👍🏻